src/components/rangeselector/button_attributes.js

```diff
@@ -0,0 +1,44 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+
+module.exports = {
+    step: {
+        valType: 'enumerated',
+        role: 'info',
+        values: ['month', 'year', 'day', 'all'],
+        dflt: 'month',
+        description: [
+            ''
+        ].join(' ')
+    },
+    stepmode: {
+        valType: 'enumerated',
+        role: 'info',
+        values: ['backward', 'to date'],
```

Comment:
An important distinction here.

With `backward`, the starting pt (i.e `range[0]`) goes back `(count * step)` in ms.

With `to date`, the starting pt (i.e `range[0]`) goes back to the *floor* of `(count * step)`. 

For example, with 

```js
{
  step: 'year',
  stepmode: 'to date', 
  count: 1
}
```

the starting range value is `Jan 01` of the last year in the time series.